### PR TITLE
rt: refactor drop logic

### PR DESCRIPTION
**Pull request by Noah-Kennedy**
This shouldn't change any functionality, but it does use ManuallyDrop to cleanup the drop logic a fair bit. This is much nicer than relying on drop order.

### CODE REVIEW

File: src/runtime/mod.rs
Comment (FrankReh) [inline src/runtime/mod.rs:130]: Interesting. This shows the unset happens after local tasks and rt are dropped. Well, if that works, it seems very clean. The order is surprising given the order of creation in new, but I guess that's the point.

**Review by FrankReh — APPROVED** [review]
I like it. Less code but more  importantly, makes reasoning about the order of things very straight forward.

File: src/runtime/mod.rs
Comment (FrankReh) [inline src/runtime/mod.rs:3]: This line is breaking CI it appears.

**Review by FrankReh — CHANGES_REQUESTED** [review]

File: src/runtime/mod.rs
Comment (Noah-Kennedy) [reply src/runtime/mod.rs:3]: Yeah, this got pulled in by mistake. I'll pull it out once I finish another PR.

Comment (Noah-Kennedy) [reply src/runtime/mod.rs]: Fixed.

**Comment by Noah-Kennedy** [comment]
@FrankReh its ready for re-review.

**Comment by FrankReh** [comment]
On my cell. Be home in an hour. 

-Frank

**Comment by FrankReh** [comment]
If ci passes now. Squash commit yourself and don’t wait for me. Code looked good before. 

-Frank

**Comment by Noah-Kennedy** [comment]
I need to wait for your approval because you requested changes.

**Review by FrankReh — APPROVED** [review]

**Comment by FrankReh** [comment]
Somehow the system deletes the branch automatically just because I squashed committed. It wasn't me. I promise. ;)

**Pull request merged**
